vm-setup/roles/v1aX_integration_test/templates/controlplane_ubuntu.yaml

```diff
@@ -128,11 +128,19 @@ spec:
       name: {{ CLUSTER_NAME }}-controlplane-0
   infrastructureRef:
     apiVersion: infrastructure.cluster.x-k8s.io/{{ CAPI_VERSION }}
+{% if CAPI_VERSION == 'v1alpha3' %}
```

Comment:
may be if CAPI_VERSION != v1alpha2 && CAPI_VERSION !=v1alpha1, since this change is here to stay I suppose. So, kind will be the same even for v1alpha4 etc.

Reply:
good point!

Reply:
Since these CR templates are only for v1a2/v1a3/v1ax,  I would rather go for for example:
```
{% if CAPI_VERSION == 'v1alpha2' %}
kind: BareMetalMachine
{% else %}
kind: Metal3Machine
{% endif %}
```
because in case of v1alpha1 we use completely different [templates ](https://github.com/metal3-io/metal3-dev-env/blob/master/scripts/v1alpha1/create_machine.sh)